### PR TITLE
Mapping from Calendar to Vaadin dayOfWeek

### PR DESCRIPTION
**Pull request by alferink**
See https://github.com/jmix-framework/jmix/issues/2795 for details

### CODE REVIEW

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/DateInternationalizationHelper.java [hunk 1]
```diff
@@ -80,4 +80,8 @@ public static int getFirstDayOfWeek(Locale locale) {
         Calendar calendar = Calendar.getInstance(locale);
         return calendar.getFirstDayOfWeek();
     }
+
+    public static int calendarDayOfWeekToVaadinDayOfWeek(int calenderDayOfWeek) {
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/DateInternationalizationHelper.java:84]: We don't need a separate method, just fix `getFirstDayOfWeek`.

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/datepicker/TypedDatePicker.java [hunk 2]
```diff
@@ -96,7 +96,8 @@ protected void initComponent() {
                 .setWeekdays(getWeekdayNames(locale))
                 .setWeekdaysShort(getShortWeekdayNames(locale))
                 .setMonthNames(getMonthNames(locale))
-                .setFirstDayOfWeek(getFirstDayOfWeek(locale)));
+                .setFirstDayOfWeek(calendarDayOfWeekToVaadinDayOfWeek(getFirstDayOfWeek(locale)))
```
Comment (glebfox) [inline jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/datepicker/TypedDatePicker.java:99]: Pay attention that not on TypedDatePicker uses `getFirstDayOfWeek` but `TypedDateTimePicker` as well

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/DateInternationalizationHelper.java (hunk 1 above)
Comment (alferink) [reply jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/DateInternationalizationHelper.java]: I have tried to do the changes you requested. Please let me know if you have other change requests.

**Review by glebfox — APPROVED** [review]

**Comment by glebfox** [comment]
Thank you for your participation

**Pull request merged**